pkg/arvo/lib/hark/store.hoon

```diff
@@ -0,0 +1,220 @@
+/-  sur=hark-store, post
+/+  resource, graph-store, group-store, chat-store
+^?
+=<  [. sur]
+=,  sur
+|%
+++  dejs
+  =,  dejs:format
+  |%
+  ++  index
+    %-  of
+    :~  graph+graph-index
+        group+group-index
+        chat+chat-index
+    ==
+  ::
+  ++  chat-index
+    %-  ot
+    :~  chat+pa
+        mention+bo
+    ==
+  ::
+  ++  group-index
+    %-  ot
+    :~  group+dejs-path:resource
+        description+so
+    ==
+  ::
+  ++  graph-index
+    %-  ot
+    :~  group+dejs-path:resource
+        graph+dejs-path:resource
+        module+so
+        description+so
+    ==
+  ::  parse date as @ud
+  ::    TODO: move to zuse
+  ++  sd
+    |=  jon=json 
+    ^-  @da
+    ?>  ?=(%s -.jon)
+    `@da`(rash p.jon dem:ag)
+
+  ::
+  ++  notif-ref
+    ^-  $-(json [@da ^index])
+    %-  ot
+    :~  time+sd
+        index+index
+    ==
+  ::
+  ++  add
+    |=  jon=json
+    [*^index *notification]
+  ::
+  ++  action
+    ^-  $-(json ^action)
+    %-  of
+    :~  seen+ul
+        archive+notif-ref
+        unread+notif-ref
+        read+notif-ref
+        add+add
+        set-dnd+bo
+    ==
+  --
+::
+++  enjs
+  =,  enjs:format
+  |%
+  ::
+  ++  update
+    |=  upd=^update
+    ^-  json
+    |^
+    %+  frond  -.upd
+    ?+  -.upd  a+~
+        %added      (added +.upd)
+        %timebox  (timebox +.upd)
+        %set-dnd  b+dnd.upd
+        %count    (numb count.upd)
+        %more     (more +.upd)
+          ?(%archive %read %unread)
```

Comment:
extremely weird spacing

Reply:
how is this weird? it's queenside, so we backdent the first hoon and have the second hoon on the next line. Missing `::` is questionable though

Reply:
In all matters whitespace I direct you to https://github.com/urbit/urbit/files/4132264/whitespace.pdf

In particular, it's _not_ queenside: kingside is when the stems are indented one stop and the bulbs are inline or indented more, queenside is when the stems are indented two stops and the bulbs are indented one stop(or inline when short), here a stem is indented _three_ and its bulb two. For proper either-side, you want all of the patterns aligned with each other; what \~does\~ want to be at three stops is the `-.upd`, so it's backdented from instead of vertically aligned to the stems.